instance/openstack.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+OpenStack - Helper functions
+"""
+
+# Imports #####################################################################
+
+import requests
+
+from novaclient.v2.client import Client as NovaClient
```

Comment:
@dachary Thanks for the advice - I haven't run into any specific issue with the API class so far, and directly using it allowed to do things like using the auto-retry features of requests & httplib3 (cf https://github.com/open-craft/opencraft/pull/1/files#diff-84b847358091a7dddf73724bcbd2b9d3R57 ), but I can see how the interface of the cli would be more bullet-proof and supported. I'll have a look at switching to it.